CMakeLists.txt

```diff
@@ -0,0 +1,551 @@
+
+# Supported AMD GPU architectures.
+set(ROCM_SUPPORTED_ARCHS "gfx908;gfx90a;gfx942;gfx1100")
+
+#
+# Supported/expected torch versions for CUDA/ROCm.
+#
+# Currently, having an incorrect pytorch version results in a warning
+# rather than an error.
+#
+# Note: the CUDA torch version is derived from pyproject.toml and various
+# requirements.txt files and should be kept consistent.  The ROCm torch
+# versions are derived from Dockerfile.rocm
+#
+set(TORCH_SUPPORTED_VERSION_CUDA "2.1.2")
+set(TORCH_SUPPORTED_VERSION_ROCM_5X "2.0.1")
+set(TORCH_SUPPORTED_VERSION_ROCM_6X "2.1.1")
+
+#
+# Loop through all supported python versions until the first suitable version
+# that has torch installed.
+#
+# Note: cmake is unable to pick the lowest supported version when multiple
+# versions are available, even with `CMAKE_FIND_PACKAGE_SORT_ORDER`.
+#
+foreach(VER ${PYTHON_SUPPORTED_VERSIONS})
+  message(STATUS "Attempting to find python ${VER} package.")
+  find_package(Python ${VER} COMPONENTS Interpreter Development.Module)
+  if (Python_FOUND)
+    # Attempt to import torch from python was just found. If torch can
+    # be imported, stop searching for other versions of python.
+    # Note: find_package(Torch) won't work here because cmake might not
+    # have the proper search path set yet.
+    execute_process(
+      COMMAND
+      "${Python_EXECUTABLE}" "-c" "import torch"
+      OUTPUT_VARIABLE PYTHON_OUT
+      RESULT_VARIABLE PYTHON_ERROR_CODE
+      ERROR_VARIABLE PYTHON_STDERR)
+
+    if(PYTHON_ERROR_CODE EQUAL 0)
+      message(STATUS
+        "Found python version ${Python_VERSION} (${Python_EXECUTABLE}).")
+      break()
+    endif()
+  endif()
+endforeach()
+
+if (NOT Python_FOUND)
+  message(FATAL_ERROR
+    "No supported version of python (with pytorch) found. "
+    "('${PYTHON_SUPPORTED_VERSIONS}')")
+endif()
+
+#
+# Run `EXPR` in python.  The standard output of python is stored in `OUT` and
+# has trailing whitespace stripped.  If an error is encountered when running
+# python, a fatal message `ERR_MSG` is issued.
+#
+macro (run_python OUT EXPR ERR_MSG)
+  execute_process(
+    COMMAND
+    "${Python_EXECUTABLE}" "-c" "${EXPR}"
+    OUTPUT_VARIABLE ${OUT}
+    RESULT_VARIABLE PYTHON_ERROR_CODE
+    ERROR_VARIABLE PYTHON_STDERR
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if(NOT PYTHON_ERROR_CODE EQUAL 0)
+    message(FATAL_ERROR "${ERR_MSG}: ${PYTHON_STDERR}")
+  endif()
+endmacro()
+
+#
+# Update cmake's `CMAKE_PREFIX_PATH` with probable torch locations.
+#
+
+# Run `EXPR` in python after importing `PKG`. Use the result of this to extend
+# `CMAKE_PREFIX_PATH` so the torch cmake configuration can be imported.
+macro (append_cmake_prefix_path PKG EXPR)
+  run_python(PREFIX_PATH
+    "import ${PKG}; print(${EXPR})" "Failed to locate ${PKG} path")
+  list(APPEND CMAKE_PREFIX_PATH ${PREFIX_PATH})
+endmacro()
+
+# Add user site-packages path to `CMAKE_PREFIX_PATH`.
+append_cmake_prefix_path("site" "site.getusersitepackages()")
```

Comment:
Why is this needed / done?

Reply:
this was leftover cruft that isn't needed anymore.